### ISSUE TITLE
Entries (Sprout Forms) report shows Name fieldtype data as "Unsupported Field"

### ISSUE THREAD

**Issue by bentilden** [barrelstrength/craft-sprout-reports#94]
### Description

After updating Sprout Forms from 3.11.2 to 3.12.2 and Sprout Reports from 1.5.1 to 1.5.6, we noticed that data captured by Sprout Forms in a Name datatype field now displays in reports as "Unsupported Field".

### Steps to reproduce

1. Create Sprout Forms form with a field of type "Name".
2. Add at least one entry.
3. Create Sprout Reports report of type "Entries (Sprout Forms)"; select the new form as the data source; save.
4. Run the report.

### Additional info

- Craft version: 3.4.24
- PHP version: 7.2.22
- Database driver & version:  MySQL 5.7.26
- Plugins & versions: Sprout Reports 1.5.6

**Comment by BenParizek** [comment]
Thanks for reporting @bentilden. We'll get that displaying again and update this thread once we do.

**Comment by tplassman** [comment]
@BenParizek,

This issue was affecting a client of our as well using the reporting functionality in Sprout Forms.

After looking at the codebase, it appears the the Phone model in Sprout Base Fields does contain a `__toString()` method, but this was not triggering the `is_string` method on line 205 of the [EntriesDataSource file](https://github.com/barrelstrength/craft-sprout-forms/blob/v3/src/integrations/sproutreports/datasources/EntriesDataSource.php#L205).  I found this functionality was discussed on the [is_string documentation page](https://www.php.net/manual/en/function.is-string.php#121637).

In order to leverage the `__toString()` method of the Sprout Base Fields [Phone model](https://github.com/barrelstrength/craft-sprout-base-fields/blob/v1/src/models/Phone.php#L62), the check on line 205 would need to include a `method_exists` call e.g.:

```php
elseif (is_string($field) or $field === null or method_exists($field, '__toString'))
```

This seemed to work to address my issue, but I did not have any other test cases to apply this to.

Hope this information helps.

Best,
Trevor

**Comment by mark-adigital** [comment]
We saw this issue too, can confirm that @tplassman 's fix worked.

**Comment by wes-lawson** [comment]
@BenParizek was this issue ever addresses? I am still receiving "Unsupported Field" for Name field types in my report. I have tested on Craft 3.5.15.1 and have upgraded to Sprout Forms 3.13.1.

Thanks,
Wes

**Comment by BenParizek** [comment]
@wes-lawson Looks like I overlooked this when I fixed some similar fields in 3.13.0.

I'll get this out in the next release, which is coming soon but I'm still working through a few migration issues. If you'd like to add this in manually before I get that out, you can see the commit here:
https://github.com/barrelstrength/craft-sprout-forms/commit/2500a5d3c5649d51d166dec301adb8c11e258375

**Comment by BenParizek** [comment]
Support in Reports for the Name field type has been added in the latest release.

I'm catching up on several support tickets right now and working on a large update to improve the migration process that has been causing a lot of issues in our plugins. There will likely be several smaller updates coming out over the next couple of weeks so, if possible, I'd recommend holding off on updating until we get Sprout Forms v4.0 out the door which should get things into a lot better shape and address a number of smaller issues all at once.
